### ISSUE TITLE
Secrecy or Vault should close on screen locking

### ISSUE THREAD

**Issue by laura-almieda** [SecrecySupportTeam/secrecy#118]
Case 1 : 
At the Secrecy main activity (Vault selection screen) screen locking (click on device power button hardware key and again unlocking the device) does not do anything.

Case 2 : 
Inside the vault, Screen locking and unlocking will returns the app to the Main Activity.

Case 3 : 
Inside the vault while viewing an image, Screen locking and unlocking does not do anything.
After unlocking the phone the encrypted image is still open. This is a problem and it should be Fix.

PS : 
I recommend that Secrecy should be closed on screen locking on the above 3 cases.

**Comment by L-Henke** [comment]
#119 fixed number 3.
We'll think about a way to exit the app instead of returning to the vault list.

**Comment by StelianMorariu** [comment]
"Case 2 : 
Inside the vault, Screen locking and unlocking will returns the app to the Main Activity."

 Considering case 3, viewing encrypted content after screen unlock is not ok - because a vault is basically a collection of encrypted files I would think the app is doing the right thing (returning to the Main activity).

My opinion is that 1&2 are not bugs. Closing the app on screen lock would be a useful feature (but I would apply it only when stealth mode is activated)

**Comment by L-Henke** [comment]
@StelianMorariu good point. Only directly closing the app when in stealth mode sounds very reasonable.

**Comment by Doplgangr** [comment]
Sorry for accidentally closing this.

stealth mode hide still needs another PR to be implemented.
